src/main/java/org/mockito/internal/matchers/CapturingMatcher.java

```diff
@@ -1,43 +1,42 @@
+
 /*
  * Copyright (c) 2007 Mockito contributors
  * This program is made available under the terms of the MIT License.
  */
 package org.mockito.internal.matchers;
 
-import org.mockito.ArgumentMatcher;
-import org.mockito.exceptions.Reporter;
-
 import java.io.Serializable;
-import java.util.LinkedList;
 import java.util.List;
 
-@SuppressWarnings("unchecked")
-public class CapturingMatcher<T> implements ArgumentMatcher<T>, CapturesArguments, VarargMatcher, Serializable {
-    
-    private final LinkedList<Object> arguments = new LinkedList<Object>();
+import org.mockito.ArgumentMatcher;
 
-    public boolean matches(Object argument) {
-        return true;
-    }    
+public class CapturingMatcher<T>
+        implements ArgumentMatcher<T>, CapturesArguments, VarargMatcher, Serializable {
+
+    private final List<T> arguments;
+    private final ArgumentMatcher<T> matcher;
+
+    public CapturingMatcher(ArgumentMatcher<T> matcher, List<T> arguments) {
+        this.matcher = matcher;
+        this.arguments = arguments;
+    }
 
     public String toString() {
         return "<Capturing argument>";
     }
 
-    public T getLastValue() {
-        if (arguments.isEmpty()) {
-            new Reporter().noArgumentValueWasCaptured();
-            return null;
-        } else {
-            return (T) arguments.getLast();
+    public void captureFrom(Object argument) {
+        if (matcher.matches(argument)) {
+            this.arguments.add((T) argument);
         }
     }
 
-    public List<T> getAllValues() {
-        return (List) arguments;
+    @Override
+    public boolean matches(Object argument) {
+        return matcher.matches(argument);
     }
 
-    public void captureFrom(Object argument) {
-        this.arguments.add(argument);
+    public static <T> CapturingMatcher<T> is(ArgumentMatcher<T> argumentMatcher, List<T> arguments) {
```

Comment:
Is that static method needed, what is the benefit of it compared to the constructor?